api/api/specs/routers.yaml

```diff
@@ -250,6 +250,40 @@ paths:
           description: "Invalid project_id or router_id"
         404:
           description: "No router versions found"
+    post:
+      tags: *tags
+      summary: "Create router version without deploying it"
+      parameters:
+        - in: "path"
+          name: "project_id"
+          description: "project id of the project of the router"
+          schema:
+            <<: *id
+          required: true
+        - in: "path"
+          name: "router_id"
+          description: "id of the router to create a new version for"
+          schema:
+            <<: *id
+          required: true
+      requestBody:
+        description: "router configuration to save"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RouterConfig"
```

Comment:
Yep, that would make the entire request body more logical, and reduces unnecessary checks (as you've pointed out below in the API function that updates routers/creates new router versions). It would also make updating routers a little less cumbersome with the SDK as users would no longer need to handle the `environment_name` and `(router_)name` attributes.

Would calling the new struct `RouterUpdateConfig` be a little clearer? I was thinking that the name would reflect its intended use to be in 'requests only'. 

And if we're introducing the new struct right now, I think we should probably change the API/UI/SDK all at once too, in order to avoid breaking anything. Or, we could do everything in a separate task. But I'm okay with doing everything right now.

Reply:
I thought we could deal with the update router API (request body change from `RouterConfig` -> `RouterVersionConfig`) later on. In this case, the suggested change should only affect the SDK (as the classes are auto-generated) and can be carried out in this MR.

As you've noted, the update router API request body change will affect parts of the API too. And in general, there are other APIs about which the same argument can be made (eg: ensembler name and type), i.e., should the update request expect fields that are not allowed to be changed. So we can capture this in the backlog and revisit at a later time.